build-system/tasks/compile.js

```diff
@@ -197,23 +197,38 @@ function compile(entryModuleFilename, outputDir,
             '|' + sourceMapBase,
         warning_level: process.env.TRAVIS ? 'QUIET' : 'DEFAULT',
       }
-    }))
-    .on('error', function(err) {
-      console./*OK*/error(err.message);
-      process.exit(1);
-    })
-    .pipe(rename(outputFilename))
-    .pipe(replace(/\$internalRuntimeVersion\$/g, internalRuntimeVersion))
-    .pipe(replace(/\$internalRuntimeToken\$/g, internalRuntimeToken))
-    .pipe(gulp.dest(outputDir))
-    .on('end', function() {
-      console./*OK*/log('Compiled', entryModuleFilename, 'to',
-          outputDir + '/' + outputFilename, 'via', intermediateFilename);
-      gulp.src(intermediateFilename + '.map')
-          .pipe(rename(outputFilename + '.map'))
-          .pipe(gulp.dest(outputDir))
-          .on('end', resolve);
-    });
+    };
+
+    if (argv.typecheck_only) {
```

Comment:
Check out the `--checks-only` command line flag. This might yield better results as it sets the correct default options for passes and might simplify this change by a lot.